### ISSUE TITLE
Subtasks is not defined

### ISSUE THREAD

**Issue by comply-online** [wekan/wekan#2115]
## Issue

**Server Setup Information**:

* Did you test in newest Wekan?: `yes, 2.01`
* For new Wekan install, did you configure root-url correctly https://github.com/wekan/wekan/wiki/Settings ? `yes`
* Wekan version: `2.01`
* Operating System: `docker/kubernetes`
* Deployment Method(snap/docker/sandstorm/mongodb bundle/source): `yaml file`
* Node Version: `alpine 10`
* MongoDB Version: `3.2.21`
* ROOT_URL environment variable http(s)://(subdomain).example.com(/suburl):

**Problem description**:

Getting `Subtasks is not defined` when creating a card and deleting it right away.

Already looked at https://github.com/wekan/wekan/issues/1986 but this isn't helpful as it was inconclusive.

- *Explain steps how to reproduce*

* Create a new user (`POST /api/users`)
* Create a new board (`POST /api/boards`)
* Create a new list (`POST /api/boards/${boardId}/lists`)
* Create a new swimlane (`POST /api/boards/${boardId}/swimlanes`)
* Create a card (using `POST /api/boards/${boardId}/lists/${listId}/cards` with `swimLaneId` coming from the previous step and `authorId` coming from user that was created above)
* Delete the card using the card id from last step (`DELETE /api/boards/${boardId}/lists/${listId}/cards/${cardId}`) (also sending `authorId` in the body)

Error shows as:

```
2019-01-21T21:30:44.314310092Z ReferenceError: Subtasks is not defined
2019-01-21T21:30:44.314367991Z     at cardRemover (models/cards.js:1254:3)
2019-01-21T21:30:44.314384691Z     at models/cards.js:1568:5
2019-01-21T21:30:44.314398791Z     at packages/simple_json-routes.js:98:9
```

**Comment by rjevnikar** [comment]
This is because of the lines
 ```
 Subtasks.remove({
    cardId: doc._id,
  });
```
in the cardRemover function in models/cards.js. Off the top of my head there are two fixes:
a) if you are deleting cardA, you loop through all the cards with cardA as a parentId and delete them. If there are nested subtasks, this could get rather loopy.
b) if you are deleting cardA, you loop through all the cards with cardA as a parentId and remove the parentID, making them top level cards - I'm not sure if this is a good idea.

To make either option work, the ability to assign a card as its own parent (#2202) would need to be addressed, or things could get messy.

**Comment by xet7** [comment]
@rjevnikar 

Do you have time to try to fix this and #2202 ?

**Comment by rjevnikar** [comment]
> @rjevnikar
> Do you have time to try to fix this and #2202 ?

I can, but it will be slow

**Comment by xet7** [comment]
Fixed already in Wekan release.
